src/Export/DynamicProductGroupService.php

```diff
@@ -121,46 +124,49 @@ public function setSalesChannel(SalesChannelEntity $salesChannelEntity): void
 
     public function warmUp(): void
     {
-        $cacheItem = $this->getCacheItem();
         $products = $this->parseProductGroups();
         if (Utils::isEmpty($products)) {
+            $this->cacheHandler->dynamicProductGroupWarmUp($this->start, $this->count);
+
             return;
         }
-        $cacheItem->set(serialize($products));
-        $cacheItem->expiresAfter(self::CACHE_LIFETIME_PRODUCT_GROUP);
-        $this->cache->save($cacheItem);
+
+        $this->cacheDynamicProductGroupTotal();
+        $this->cacheDynamicProductOffset($products);
+
+        $this->cacheHandler->dynamicProductGroupWarmUp($this->start, $this->count);
     }
 
-    public function isWarmedUp(): bool
+    public function isOffsetWarmedUp(): bool
     {
-        if ($this->start === 0) {
-            return false;
-        }
-
-        $cacheItem = $this->getCacheItem();
-        if ($cacheItem && $cacheItem->isHit()) {
-            $cacheItem->expiresAfter(self::CACHE_LIFETIME_PRODUCT_GROUP);
-            $this->cache->save($cacheItem);
+        return $this->cacheHandler->isCacheWarmedUp($this->start);
+    }
 
-            return true;
-        }
+    public function isDynamicProductGroupsCached(): bool
+    {
+        return $this->cacheHandler->isDynamicProductGroupsCached();
+    }
 
-        return false;
+    public function getDynamicProductGroupTotal()
```

Comment:
What is the purpose of adding a method that will only call a single other method? This bloats the class, which already does so many things even more.

Reply:
Please remove these single-line methods and simply call the `cacheHandler` methods directly.

Reply:
This is done because the export controller is using the dynamic product service for a reason and the cache handling is being done in that service. These one-line methods are being called from the export controller and the export controller does not use the cache handler directly.